### ISSUE TITLE
Mods leaderboard doesn't show scores with only selected mods

### ISSUE THREAD

**Issue by Cynplytholowazy** [ppy/osu-web#4674]
Related: ppy/osu#5361 

The web checks if the score contains one of the mods selected, not all of them, It should behave like it will look for scores that matches all of the combination of mods.

![image](https://cdn.discordapp.com/attachments/540371143643758592/600541780080853021/Screenshot_2019-07-16-12-18-12-391_org.mozilla.fennec_aurora.png)

If that is applied, not being able to select conflicting mods needs to be applied too.

![image](https://cdn.discordapp.com/attachments/188630616286167041/600056726313500677/unknown.png)

**Comment by peppy** [comment]
Does anyone disagree with this change? It would result in no longer being able to search inclusively (ie. "i want to see all scores with hidden +anything").

**Comment by Joehuu** [comment]
Duplicate of #2950.

**Comment by Cynplytholowazy** [comment]
Not duplicate because this issue is about choosing more than 1 mods and resulting in the site looking for either one that is present in the site. For example, in the above screenshot, choosing EZ and HD will result in web displaying scores that include either EZ or HD.

Also, I probably didn't word that very well but what I meant is that if 2 mods are chosen (like EZ and HD in the above case), it will show scores that have [EZ + HD + anything]. If things get fixed this way it's unrelated to #2950.

**Comment by FullTablet** [comment]
What about making each mod button have 3 states?
Default: Show the score no matter if it contains the mod or not. The button is faded.
Must Include: Don't show scores that don't contain the mod. The button is highlighted.
Don't Include: Don't show scores that contain the mod. The button is crossed out.

For DT/NC, consider scores with NC as having DT enabled as well, unless DT is in "Don't Include" status and NC is in "Must Include" status, in this case NC scores should still show (equivalent as having DT in Default state and NC in Must Include status).

I think that system is not very complex and it covers most use cases.

**Comment by Cynplytholowazy** [comment]
Pretty sure peppy explained he don't want the UI to be too complicated in #2950.

**Comment by FullTablet** [comment]
I don't consider a 3-state system for each button complicated.

If you wanted to reduce complexity, you could remove the "Don't Include" status. This makes the system equivalent to what was proposed in the original post (if I understood it correctly, peppy seems to have got another interpretation). I think that system would still be more useful and intuitive than the current system.

**Comment by Cynplytholowazy** [comment]
That might require a design rework too. Also if that is to be practiced it means 2 indexing methods are to be working simultaneously, which may increase server / database load. But let's see how devs see first I guess,

**Comment by abraker95** [comment]
I agree that the 3-state system is not complicated at all. I also think this offers a better experience because the player will have better flexibility in choosing what kind of scores they want displayed - specifically ones that are more relevant to them. Suppose the player wants to see all the DT scores, but without HR because the player doesn't play or care about HR. You can't do that currently.

**Comment by peppy** [comment]
Let me rephrase my question: is the current behaviour useful to anyone? Or should we just switch it to the requested one?

**Comment by cl8n** [comment]
the requested behavior seems a lot more useful, it'd match the in-game mod leaderboards too

**Comment by peppy** [comment]
Let's change this.
